include/opendht/peer_discovery.h

```diff
@@ -56,79 +43,68 @@ class OPENDHT_PUBLIC PeerDiscovery
     /**
      * startDiscovery - Listen
     */
-    void startDiscovery(PeerDiscoveredCallback callback, in_port_t port_to_avoid);
+    void startDiscovery(PeerDiscoveredCallback callback, const dht::InfoHash &nodeId);
 
     /**
      * startPublish - Send
     */
-    void startPublish(dht::InfoHash nodeId, in_port_t port_to_send);
+    void startPublish(const dht::InfoHash &nodeId, in_port_t port_to_send);
 
     /**
      * Send socket procudure start - one time sender
     */
-    void sender_oneTimeShoot(uint8_t * data_n, in_port_t port_to_send);
+    void publishOnce(const dht::InfoHash &nodeId, in_port_t port_to_send);
 
     /**
      * Listener socket procudure start - one time Listen
     */
-    uint32_t listener_oneTimeShoot();
+    uint32_t discoveryOnce();
 
     /**
      * Thread Stopper
     */
-    void stop(bool listenorsend){ 
+    void stop(){ 
         
-        continue_to_run_setter(false);
         cv_.notify_one();
-        if(listenorsend) {
-
-            close(stop_readfd_);
-            if (stop_writefd_ != -1) {
-                if (write(stop_writefd_, "\0", 1) == -1) {
-                    perror("write");
-                }
+        running_ = false;
+        close(stop_readfd_);
```

Comment:
if you close stop_readfd here you won't be able to use it so write(stop_writefd...) afterward is useless. close(stop_readfd_) is already called at the right place at the end of the thread.